floodmodeller_api/hydrology_plus/hydrograph.py

```diff
@@ -0,0 +1,89 @@
+"""
+Flood Modeller Python API
+Copyright (C) 2024 Jacobs U.K. Limited
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see https://www.gnu.org/licenses/.
+
+If you have any query about this program or this License, please contact us at support@floodmodeller.com or write to the following
+address: Jacobs UK Limited, Flood Modeller, Cottons Centre, Cottons Lane, London, SE1 2QG, United Kingdom.
+"""
+
+import pandas as pd
+
+from floodmodeller_api._base import FMFile
+from floodmodeller_api.util import handle_exception
+
+
+class HydrographPlusExport(FMFile):
+    """Class to handle the output of Hydrology +.
+    Args:
+        The csv file produced by Hydrology + in Flood Modeller
+
+    Output:
+        Initiates 'HydrographPlus' object
+        The event/s needed to run simulations in Flood Modeller
+    """
+
+    _filetype: str = "HydrographPlusExport"
+    _suffix: str = ".csv"
+
+    @handle_exception(when="read")
+    def __init__(self, csv_file_path, from_json: bool = False):
+        if from_json:
+            return
+        FMFile.__init__(self, csv_file_path)
+        self._read()
+
+    def _read(self):
+        self.data_file = pd.read_csv(self._filepath)
+        self.metadata = self._get_metadata()
+        self.data_flows = self._get_df_hydrographs_plus()
+
+    def _get_metadata(self) -> dict[str, str]:
+        """To extract the metada from the hydrology + results"""
+        metadata_row_index = self.data_file.index[
+            self.data_file.apply(lambda row: row.str.contains("Return Period")).any(axis=1)
+        ][0]
+        metadata_df = self.data_file.iloc[:metadata_row_index, 0:1]
+
+        return {row.split("=")[0]: row.split("=")[1] for row in metadata_df.iloc[:, 0]}
+
+    def _get_df_hydrographs_plus(self) -> pd.DataFrame:
```

Comment:
I think this function is too complex for a simple task, I think it would be best to reload the csv based on the info found in the metadata method. 
e.g. if you set the metadata_row_index as a stored variable `self.metadata_row_index` then this whole function can be replaced with `pd.read_csv(self._filepath, skiprows=metadata_row_index + 5, index_col=0)` which will allow pandas to automatically handle the dtypes and index

Reply:
I believe that we shouldn't use this function that you propose.  What would happen if in the future the number of rows change by adding more metadata? in that case, the index would change.  that is why I preferred to use the string "Time (hours)" than index.